chips/nrf52/src/comp.rs

```diff
@@ -0,0 +1,301 @@
+use kernel::common::cells::OptionalCell;
+use kernel::common::registers::{
+    register_bitfields, register_structs, ReadOnly, ReadWrite, WriteOnly,
+};
+use kernel::common::StaticRef;
+use kernel::hil::analog_comparator;
+use kernel::ReturnCode;
+
+/// Representation of an AC channel on the NRF52.
+pub struct AcChannel {
+    chan_num: u32,
+}
+
+// Only one channel, code taken from sam4l implementation
+#[derive(Copy, Clone, Debug)]
+#[repr(u8)]
+enum Channel {
+    AC0 = 0x00,
+}
+
+/// Initialization of an AC channel.
+impl AcChannel {
+    /// Create a new AC channel.
+    ///
+    /// - `channel`: Channel enum representing the channel number
+    const fn new(channel: Channel) -> AcChannel {
+        AcChannel {
+            chan_num: (channel as u32) & 0x0F,
+        }
+    }
+}
+
+pub static mut CHANNEL_AC0: AcChannel = AcChannel::new(Channel::AC0);
+
+// Base address 0x40013000
+register_structs! {
+    CompRegisters {
+        // TASK REGISTERS
+        // Trigger task by writing 1
+        // start comparator. Needed before comparator does anything.
+        // After start, triggers events_ready
```

Comment:
Can you update all of these to be doc comments `///` rather than `//`, that way we get documentation generated for them (e.g. https://5e23ab06b4899600087c8c17--docs-tockosorg.netlify.com/nrf52/i2c/struct.twimregisters#structfield.events_suspended)

Reply:
Fixed.